catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/TemplateTransformer.java

```diff
@@ -233,4 +233,47 @@ private static String retrieveOwnerIfPresent(Metacard inputMetacard) {
     return ImmutableMap.of(
         Security.ACCESS_INDIVIDUALS, accessIndividuals, Security.ACCESS_GROUPS, accessGroups);
   }
+
+  private static Map<String, Object> collapseFilterFunctions(Map<String, Object> filter) {
+    return transformFilterFunctions(
+        filter, JsonFunctionTransform::canApplyFromFunction, JsonFunctionTransform::fromFunction);
+  }
+
+  /**
+   * Transform a filter JSON structure based upon the invocations provided.
+   *
+   * @param filter original filter JSON map.
+   * @param shouldDoTransform invocation on transform; calls appropriate boolean check.
+   * @param howToDoTransform invocation on transform; calls the actual transform itself.
+   * @return transformed filter JSON map.
+   */
+  private static Map<String, Object> transformFilterFunctions(
+      Map<String, Object> filter,
+      BiFunction<JsonFunctionTransform, Map<String, Object>, Boolean> shouldDoTransform,
+      BiFunction<JsonFunctionTransform, Map<String, Object>, Map<String, Object>>
+          howToDoTransform) {
+    if ("AND".equals(filter.get("type")) || "OR".equals(filter.get("type"))) {
+      return ImmutableMap.<String, Object>builder()
+          .put("type", filter.get("type"))
+          .put(
+              "filters",
+              ((List<Map<String, Object>>) filter.get("filters"))
+                  .stream()
+                  // This will also be a BiFunction param when filter func expansions are needed
+                  // Hard-coded to "collapseFilterFunctions" for now
+                  .map(TemplateTransformer::collapseFilterFunctions)
+                  .collect(Collectors.toList()))
+          .build();
+    }
+    // Note that, depending upon the context:
```

Comment:
This comment seems to be speaking to the implementation of other classes; it doesn't seem maintainable if the other classes change.  I'm not sure we need this